From 98a7924ac2213e33ea70df77a6e33320e5316b78 Mon Sep 17 00:00:00 2001
From: rselvaraj2 <rselvaraj2@maxlinear.com>
Date: Wed, 14 Aug 2024 17:30:12 +0800
Subject: [PATCH] WLANRTSYS-83979 WDS connection doesn't work in 11b and 11an

Issue:
        WDS Connection doesn't work in non-11ax modes and observing hostapd segfault.

Root Casue:
        In WDS Flow trying to access HE Capabilities which is NULL.

Fix:
        Removing redunant code which isn't required.

FIXUP to WLANRTSYS-63677 ML WDS configuration and association
---
 src/ap/mxl_wds.c | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/src/ap/mxl_wds.c b/src/ap/mxl_wds.c
index 540756ca6..7be3ed5ea 100644
--- a/src/ap/mxl_wds.c
+++ b/src/ap/mxl_wds.c
@@ -320,23 +320,6 @@ void mxl_wds_event_connect(struct hostapd_data *hapd,
 #endif /* CONFIG_IEEE80211BE */
 
 #ifdef CONFIG_IEEE80211BE_MXL_MLO
-	/* As part of EHT open source kernel sync, EHT-MCS map for BW<=80 is sent
-	 * in EHT capability IE for VHT20 in AP mode. This is inline with the standard as
-	 * EHT-MCS map for 20 MHz only is applicable for station mode. Hence for VHT20 mode,
-	 * derive MCS0-7 and MCS8-9 from MCS0-9 information of peer beacons.
-	 */
-	if (!(sta->he_capab->he_phy_capab_info[0] & (HE_PHY_CAP0_CHANNEL_WIDTH_SET_B0 |
-					HE_PHY_CAP0_CHANNEL_WIDTH_SET_B1 |
-					HE_PHY_CAP0_CHANNEL_WIDTH_SET_B2 |
-					HE_PHY_CAP0_CHANNEL_WIDTH_SET_B3))) {
-		if (sta->eht_capab_len) {
-			sta->eht_capab->optional[3] = sta->eht_capab->optional[2];
-			sta->eht_capab->optional[2] = sta->eht_capab->optional[1];
-			sta->eht_capab->optional[1] = sta->eht_capab->optional[0];
-			sta->eht_capab_len++;
-		}
-	}
-
 	if (hapd->conf->mxl_bss_conf.mlo_enable && (sta->mxl_sta_info.ext_flags & (WLAN_STA_MULTILINK))) {
 		is_both_ml_sta_assoc_received = false;
 		memcpy_s(hapd->mxl_data.sibling_peer_addr, ETH_ALEN, wds_sta->mac_addr, ETH_ALEN);
-- 
2.43.0

